File: C++/1509. Minimum Difference Between Largest and Smallest Value in Three Moves.cpp
class Solution {
public:
    int minDifference(vector<int>& nums) {
        int n=nums.size();
        if(n<=3)return 0;
        sort(nums.begin(),nums.end());
        int x=nums[n-4]-nums[0];
        int y=nums[n-1]-nums[3];
        int z=nums[n-3]-nums[1];
        int w=nums[n-2]-nums[2];
        return min({x,y,z,w});
    }
};

File: JAVA/1509. Minimum Difference Between Largest and Smallest Value in Three Moves.java
class Solution {
    public int minDifference(int[] nums) {
        int n = nums.length;
        int result = Integer.MAX_VALUE;
        if (n <= 4) return 0;
        Arrays.sort(nums);
        for (int i = n - 4; i < n; i++){
            result = Math.min(result, nums[i] - nums[i - n + 4]);
        }
        return result;
    }
}
